internal/p2p/router.go

```diff
@@ -202,6 +202,7 @@ type Router struct {
 	chDescs []*ChannelDescriptor
 
 	nodeInfoProducer func() *types.NodeInfo
+	chainID          string
```

Comment:
This is in `NodeInfo` as `Network`, do we need it twice?

Reply:
in practice calling `nodeInfoProducer` takes a lock on the node, and I'd like to (eventually) get rid of the dependency on it.